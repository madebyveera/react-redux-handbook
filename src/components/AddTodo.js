import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { addTodo } from "../store/actions";

const AddTodo = (props) => {
  const { nextTodoId, text, dispatch } = props;
  const [todoText, setTodoText] = React.useState(text);
  return (
    <div className="container">
      <form>
        <div className="form-group m-4">
          <div className="input-group">
            <input
              className="form-control"
              placeholder="Add Todo"
              value={todoText}
              onChange={event => setTodoText(event.target.value)}
            />
            <div className="input-group-append">
              <button
                type="button"
                className="btn btn-primary"
                onClick={() => dispatch(addTodo({index: nextTodoId, text: todoText}))}
              >
                Add Todo
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
  );
};

AddTodo.propTypes = {
  nextTodoId: PropTypes.number,
  text: PropTypes.string,
  dispatch: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  nextTodoId: state.todos ? state.todos.length + 1 : 1,
  text: "Use Redux"
});

export default connect(mapStateToProps)(AddTodo);
