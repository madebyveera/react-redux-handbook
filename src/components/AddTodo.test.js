import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { addTodo } from "./../store/actions";
import AddTodo from "./AddTodo";
const mockStore = configureStore([]);
describe("My Connected React-Redux Component", () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      todos: []
    });
    store.dispatch = jest.fn();
    component = renderer.create(
      <Provider store={store}>
        <AddTodo />
      </Provider>
    );
  });
  it("should render with given state from Redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("should dispatch an action on button click", () => {
    renderer.act(() => {
      component.root.findByType("button").props.onClick();
    });
    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(addTodo({index: 1, text: "Use Redux"}));
  });
});
