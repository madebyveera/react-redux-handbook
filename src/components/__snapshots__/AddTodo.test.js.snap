// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div
  className="container"
>
  <form>
    <div
      className="form-group m-4"
    >
      <div
        className="input-group"
      >
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Add Todo"
          value="Use Redux"
        />
        <div
          className="input-group-append"
        >
          <button
            className="btn btn-primary"
            onClick={[Function]}
            type="button"
          >
            Add Todo
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;
